qemu/tests/ovs_enslave_port.py

```diff
@@ -0,0 +1,65 @@
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Enslave a port which is already a member of ovs into another
+
+    1) Check current ovs bridge and list the ports in it
+    2) Create a new ovs bridge
+    3) Enslave one port which is a member of current ovs bridge into the new one
+    4) Delete the new ovs bridge
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_ovs_ports(name):
+        """
+        Get ovs ports
+
+        :param name: ovs bridge name
+        """
+        cmd = "ovs-vsctl list-ports %s" % name
+        return process.getoutput(cmd, shell=True).splitlines()
+
+    def create_ovs_bridge(name):
+        """
+        Create ovs bridge
+
+        :param name: ovs bridge name
+        """
+        if name in process.getoutput("ovs-vsctl list-br"):
+            cmd = "ovs-vsctl del-br %s" % name
+            process.system_output(cmd, shell=True)
+        cmd = "ovs-vsctl add-br %s" % name
+        process.system_output(cmd, shell=True)
+
+    def enslave_port(bridge, port):
+        """
+        Enslave port to bridge
+
+        :param bridge: ovs bridge name
+        :param port: port to enslave to the bridge
+        """
+        cmd = "ovs-vsctl add-port %s %s" % (bridge, port)
+        process.system_output(cmd, shell=True)
+
+    netdst = params.get("netdst")
+    if netdst not in process.getoutput("ovs-vsctl list-br"):
+        test.cancel("%s isn't an openvswith bridge" % netdst)
```

Comment:
We use test.skip instead of test.cancel in case level(inside case `run` function). Thanks.

Reply:
test.skip is deprecated for some time and now we use test.cancel instead.

Reply:
> test.skip is deprecated for some time and now we use test.cancel instead.

@xiywang You are right, my mistake, what we prefer is `test.cancel`, sorry for that.